### PR TITLE
Add gfp_abs_series sqrt functions and deal with char 2

### PR DESCRIPTION
**Pull request by wbhart**
I think this is ok without flint-2.9 but let's wait to see what CI says. Edit: looks like this can be merged without waiting for anything. I decided to deal with this in Nemo rather than Flint as the latter is not really practical.

### CODE REVIEW

**Comment by wbhart** [comment]
Test failures appear to be doctests unrelated to this PR.

File: test/flint/gfp_abs_series-test.jl [hunk 1]
```diff
@@ -376,6 +376,36 @@ end
...
+
+         @test q^2 == s
+
+         f1, s1 = issquare_with_sqrt(s)
+
+         @test f1 && s1^2 == s
+
+         if s*x != 0
+            @test_throws ErrorException sqrt(s*x)
+         end
```
Comment (tthsqe12) [inline test/flint/gfp_abs_series-test.jl:406]:
```suggestion
         end

         if mod(p, 4) == 3
            @test !issquare(-one(S))
         end
```

**Review by tthsqe12 — CHANGES_REQUESTED** [review]

File: test/flint/gfp_abs_series-test.jl (hunk 1 above)
Comment (tthsqe12) [reply test/flint/gfp_abs_series-test.jl:406]: for test coverage :)

Comment (tthsqe12) [reply test/flint/gfp_abs_series-test.jl:406]: other than that, fine by me

**Pull request merged**